modules/host/negotiaterenewcontract.go

```diff
@@ -0,0 +1,315 @@
+		return modules.WriteNegotiationRejection(conn, err)
+	}
+	// The host indicates acceptance, then sends the new parents, inputs, and
+	// outputs to the transaction.
+	err = modules.WriteNegotiationAcceptance(conn)
+	if err != nil {
+		return err
+	}
+	err = encoding.WriteObject(conn, newParents)
+	if err != nil {
+		return err
+	}
+	err = encoding.WriteObject(conn, newInputs)
+	if err != nil {
+		return err
+	}
+	err = encoding.WriteObject(conn, newOutputs)
+	if err != nil {
+		return err
+	}
+
+	// The renter will send a negotiation response, followed by transaction
+	// signatures for the file contract transaction in the case of acceptance.
+	// The transaction signatures will be followed by another transaction
+	// signature to sign the no-op file contract revision associated with the
+	// new file contract.
+	err = modules.ReadNegotiationAcceptance(conn)
+	if err != nil {
+		return err
+	}
+	var renterTxnSignatures []types.TransactionSignature
+	var renterRevisionSignature types.TransactionSignature
+	err = encoding.ReadObject(conn, &renterTxnSignatures, modules.NegotiateMaxTransactionSignatureSize)
+	if err != nil {
+		return err
+	}
+	err = encoding.ReadObject(conn, &renterRevisionSignature, modules.NegotiateMaxTransactionSignatureSize)
+	if err != nil {
+		return err
+	}
+
+	// The host adds the renter transaction signatures, then signs the
+	// transaction and submits it to the blockchain, creating a storage
+	// obligation in the process. The host's part is now complete and the
+	// contract is finalized, but to give confidence to the renter the host
+	// will send the sigantures so that the renter can immediately have the
+	// completed file contract.
+	//
+	// During finalization the signatures sent by the renter are all checked.
+	hostTxnSignatures, hostRevisionSignature, err := h.managedFinalizeContract(txnBuilder, renterPK, renterTxnSignatures, renterRevisionSignature)
+	if err != nil {
+		return modules.WriteNegotiationRejection(conn, err)
+	}
+	err = modules.WriteNegotiationAcceptance(conn)
+	if err != nil {
+		return err
+	}
+	// The host sends the transaction signatures to the renter, followed by the
+	// revision signature. Negotiation is complete.
+	err = encoding.WriteObject(conn, hostTxnSignatures)
+	if err != nil {
+		return err
+	}
+	return encoding.WriteObject(conn, hostRevisionSignature)
+}
+
+// managedVerifyRenewedContract checks that the contract renewal matches the
+// previous contract and makes all of the appropriate payments.
+func (h *Host) managedVerifyRenewedContract(so *storageObligation, txnSet []types.Transaction, renterPK crypto.PublicKey) error {
+	// Check that the transaction set is not empty.
+	if len(txnSet) < 1 {
+		return errEmptyFileContractTransactionSet
+	}
+	// Check that the transaction set has a file contract.
+	if len(txnSet[len(txnSet)-1].FileContracts) < 1 {
+		return errNoFileContract
+	}
+
+	h.mu.RLock()
+	blockHeight := h.blockHeight
+	externalSettings := h.externalSettings()
```

Comment:
since these were read earlier in `managedRPCRenewContract`, maybe pass them as an arg?